Add documentation on how to deploy DCA on EKS Fargate (#11550)
DataDog/integrations-core: eks_fargate/README.md

**Note**: Add your desired `kube_cluster_name:<CLUSTER_NAME>` to the list of `DD_TAGS` to ensure your metrics are tagged by your desired cluster. You can append additional tags here as space separated `<KEY>:<VALUE>` tags.

> #### Running the Cluster Agent / Cluster Checks Runner

Reviewer: ```suggestion
#### Running the Cluster Agent or the Cluster Checks Runner
```

> Running the Cluster Agent is recommended as it is required for a number of features: events collection, Kubernetes resources view, cluster checks, etc.

Reviewer 2: would be nice to link the corresponding docs for `events collection`, `Kubernetes resources view`, `cluster checks`
https://docs.datadoghq.com/agent/kubernetes/?tab=helm#event-collection
https://docs.datadoghq.com/infrastructure/livecontainers/#kubernetes-resources-view 
https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/#overview

Reviewer: Can you add add the reference links to the bottom of the doc?

[29]: https://docs.datadoghq.com/agent/kubernetes/?tab=helm#event-collection
[30]: https://docs.datadoghq.com/infrastructure/livecontainers/#kubernetes-resources-view
[31]: https://docs.datadoghq.com/agent/cluster_agent/clusterchecks/#overview

```suggestion
Datadog recommends you run the Cluster Agent to access features such as events collection, Kubernetes resources view, and cluster checks.
```

> When using EKS Fargate, two scenarios are possible depending if the EKS cluster is runnning mixed workloads (Fargate/non-Fargate) or not.

Reviewer: ```suggestion
When using EKS Fargate, there are two possible scenarios depending on whether or not the EKS cluster is running mixed workloads (Fargate/non-Fargate).
```

> If the EKS cluster runs Fargate and non-Fargate workloads **AND** you want to monitor non-Fargate workload through Node Agent DaemonSet, the Cluster Agent/Cluster Checks Runner can be easily be added to this deployment.

Reviewer: ```suggestion
If the EKS cluster runs Fargate and non-Fargate workloads, and you want to monitor the non-Fargate workload through Node Agent DaemonSet, add the Cluster Agent/Cluster Checks Runner to this deployment.
```

> Follow the [Cluster Agent Setup][25] documentation to deploy it.

Reviewer: ```suggestion
For more information, see the [Cluster Agent Setup][25].
```

> One specifity is that the Cluster Agent token must be reachable from the Fargate tasks you want to monitor, which is not the case by default using our Helm Chart or Datadog Operator as it creates a secret in the target namespace.

Reviewer: ```suggestion
The Cluster Agent token must be reachable from the Fargate tasks you want to monitor. If you are using the Helm Chart or Datadog Operator, this is not reachable by default because a secret in the target namespace is created.
```

> Two options are available to allow this to work properly:
> - Use an hardcoded token value (`clusterAgent.token` in Helm, `credentials.token` in the Datadog Operator): convenient but less secure.
> - Use a manually created secret (`clusterAgent.tokenExistingSecret` in Helm, not available in Datadog Operator) and replicate it in all namespaces where Fargate tasks need to be monitored: secure but requires extra operations.

Reviewer: ```suggestion
You have two options for this to work properly:

* Use an hardcoded token value (`clusterAgent.token` in Helm, `credentials.token` in the Datadog Operator); convenient, but less secure.
* Use a manually-created secret (`clusterAgent.tokenExistingSecret` in Helm, not available in the Datadog Operator) and replicate it in all namespaces where Fargate tasks need to be monitored; secure, but requires extra operations.
```

> If the EKS cluster runs only Fargate workloads, a standalone Cluster Agent deployment is necessary.
> This can be achieved with the following Helm `values.yaml`:

Reviewer: ```suggestion
If the EKS cluster runs only Fargate workloads, you need a standalone Cluster Agent deployment. 

Use the following Helm `values.yaml`:
```

```yaml
datadog:
  apiKey: <YOUR_DATADOG_API_KEY>
  clusterName: <CLUSTER_NAME>
agents:
  enabled: false
clusterAgent:
  enabled: true
  replicas: 2
```

> The same specifity as above regarding management of Cluster Agent token is applicable.

Reviewer: ```suggestion
**Note:** Be just as specific with managing your Cluster Agent token.
```

> Finally, in both cases the Datadog Agent sidecar manifest needs to be changed to allow communication with the Cluster Agent:

Reviewer: ```suggestion
In both cases, you need to change the Datadog Agent sidecar manifest in order to allow communication with the Cluster Agent:
```

```yaml
       env:
        - name: DD_CLUSTER_AGENT_ENABLED
          value: "true"
        - name: DD_CLUSTER_AGENT_AUTH_TOKEN
          value: <hardcoded token value> # Use valueFrom: if you're using a secret
        - name: DD_CLUSTER_AGENT_URL
          value: https://<CLUSTER_AGENT_SERVICE_NAME>.<CLUSTER_AGENT_SERVICE_NAMESPACE>.svc.cluster.local:5005
        - name: DD_ORCHESTRATOR_EXPLORER_ENABLED # Required to get Kubernetes resources view

>           value: "true"

Reviewer 2: `DD_CLUSTER_NAME` is also required for the orchestrator explorer

**Note**: Add your desired `kube_cluster_name:<CLUSTER_NAME>` to the list of `DD_TAGS` to ensure your metrics are tagged by your desired cluster. You can append additional tags here as space separated `<KEY>:<VALUE>` tags.

#### Running the Cluster Agent or the Cluster Checks Runner

> Datadog recommends you run the Cluster Agent to access features such as [events collection][29], [Kubernetes resources view][30], and [cluster checks][31].
> When using EKS Fargate, there are two possible scenarios depending on whether or not the EKS cluster is running mixed workloads (Fargate/non-Fargate).

Reviewer: ```suggestion
Datadog recommends you run the Cluster Agent to access features such as [events collection][29], [Kubernetes resources view][30], and [cluster checks][31].

When using EKS Fargate, there are two possible scenarios depending on whether or not the EKS cluster is running mixed workloads (Fargate/non-Fargate).
```

> If the EKS cluster runs Fargate and non-Fargate workloads, and you want to monitor the non-Fargate workload through Node Agent DaemonSet, add the Cluster Agent/Cluster Checks Runner to this deployment.
> For more information, see the [Cluster Agent Setup][25].

Reviewer: ```suggestion
If the EKS cluster runs Fargate and non-Fargate workloads, and you want to monitor the non-Fargate workload through Node Agent DaemonSet, add the Cluster Agent/Cluster Checks Runner to this deployment. For more information, see the [Cluster Agent Setup][25].
```

The Cluster Agent token must be reachable from the Fargate tasks you want to monitor. If you are using the Helm Chart or Datadog Operator, this is not reachable by default because a secret in the target namespace is created.

> You have two options for this to work properly:

Reviewer: ```suggestion

You have two options for this to work properly:
```

* Use an hardcoded token value (`clusterAgent.token` in Helm, `credentials.token` in the Datadog Operator); convenient, but less secure.
* Use a manually-created secret (`clusterAgent.tokenExistingSecret` in Helm, not available in the Datadog Operator) and replicate it in all namespaces where Fargate tasks need to be monitored; secure, but requires extra operations.

If the EKS cluster runs only Fargate workloads, you need a standalone Cluster Agent deployment. 

> Use the following Helm `values.yaml`:

Reviewer: ```suggestion
Use the following Helm `values.yaml`:

```

```yaml
datadog:
  apiKey: <YOUR_DATADOG_API_KEY>
  clusterName: <CLUSTER_NAME>
agents:
  enabled: false
clusterAgent:
  enabled: true
  replicas: 2
```

**Note:** Be just as specific with managing your Cluster Agent token.

> In both cases, you need to change the Datadog Agent sidecar manifest in order to allow communication with the Cluster Agent:

Reviewer: ```suggestion
In both cases, you need to change the Datadog Agent sidecar manifest in order to allow communication with the Cluster Agent:

```

Datadog Agent v6.19+ supports live processes in the EKS Fargate integration. Live processes appear on the [Processes][20] page. To enable live processes, [enable shareProcessNamespace in the pod spec][21].

### Kubernetes resources view

> Collecting Kubernetes requires a [Cluster Agent setup][#running-the-cluster-agent-cluster-checks-runner]

Reviewer: Updated the H4 above so that the dash doesn't bork the anchor link.

```suggestion
To collect Kubernetes resource views, you need a [Cluster Agent setup][#running-the-cluster-agent-or-the-cluster-checks-runner].
```

## Events collection

> To collect events from your AWS EKS Fargate API server, run a [Datadog Cluster Agent within your EKS cluster][#running-the-cluster-agent-cluster-checks-runner] and [Enable Event collection for your Cluster Agent][19].

Reviewer: ```suggestion
To collect events from your AWS EKS Fargate API server, run a [Datadog Cluster Agent within your EKS cluster][#running-the-cluster-agent-or-the-cluster-checks-runner] and [Enable Event collection for your Cluster Agent][19].
```
